### PR TITLE
Fix schemas

### PR DESCRIPTION
**Pull request by fabricio-aguiar**
closes #5311
https://pulp.plan.io/issues/5311

Please be sure you have read our documentation on creating PRs:
https://docs.pulpproject.org/en/3.0/nightly/contributing/pull-request-walkthrough.html

### CODE REVIEW

File: pulpcore/app/viewsets/upload.py [hunk 1]
```diff
@@ -60,7 +60,7 @@ def get_serializer_class(self):
 
     def get_serializer_context(self):
         context = super().get_serializer_context()
-        if self.action == 'update':
+        if self.action == 'update' and self.kwargs.get("pk"):
```
Comment (bmbouter) [inline pulpcore/app/viewsets/upload.py]: Is this related to the 500? I'm just curious.

File: pulpcore/app/viewsets/base.py [hunk 2]
```diff
@@ -8,7 +8,7 @@
 from drf_yasg.utils import swagger_auto_schema
 from rest_framework import viewsets
 from rest_framework.generics import get_object_or_404
-from rest_framework.schemas import AutoSchema
+from rest_framework.schemas.openapi import AutoSchema
```
Comment (bmbouter) [inline pulpcore/app/viewsets/base.py]: I can see how this fixes the error.

File: pulpcore/app/viewsets/upload.py (hunk 1 above)
Comment (fabricio-aguiar) [reply pulpcore/app/viewsets/upload.py]: No, it is for fixing this:
```
AssertionError: Expected view UploadViewSet to be called with a URL keyword argument named "pk". Fix your URL conf, or set the `.lookup_field` attribute on the view correctly.
```

Comment (bmbouter) [reply pulpcore/app/viewsets/upload.py]: Makes sense. Thanks!
